scripts/capture-console.py

```diff
@@ -0,0 +1,300 @@
+
+    # XOLite mode options
+    parser.add_argument('--host',
+                        help=argparse.SUPPRESS)  # Hidden - shown in epilog
+    parser.add_argument('--vm-uuid',
+                        help=argparse.SUPPRESS)  # Hidden - shown in epilog
+    parser.add_argument('--user',
+                        help=argparse.SUPPRESS)  # Hidden - shown in epilog
+    parser.add_argument('--password',
+                        help=argparse.SUPPRESS)  # Hidden - shown in epilog
+
+    # XOA mode options
+    parser.add_argument('--cookie',
+                        help=argparse.SUPPRESS)  # Hidden - shown in epilog
+
+    # Common options
+    parser.add_argument('--output-dir', dest='output_dir',
+                        help=argparse.SUPPRESS)  # Hidden - shown in epilog
+    parser.add_argument('--verbose', '-v', action='store_true',
+                        help=argparse.SUPPRESS)  # Hidden - shown in epilog
+
+    args = parser.parse_args()
+
+    # Configure logging based on verbose flag
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG, format='%(message)s')
+    else:
+        logging.basicConfig(level=logging.WARNING, format='%(message)s')
+
+    # Determine mode and validate arguments
+    if args.cookie:
+        # XOA mode: cookie-based authentication
+        if not args.url_or_filename:
+            print("Error: WebSocket URL required for XOA mode")
+            print()
+            print("Usage: python3 capture-console.py <websocket_url> [output_file] --cookie <cookie>")
+            sys.exit(1)
+
+        ws_url = args.url_or_filename
+        output_file = args.filename if args.filename else "screenshot.png"
+        cookie = args.cookie
+        logging.info("Using XOA mode (cookie-based authentication)")
+
+    elif args.host and args.vm_uuid:
+        # XOLite mode: XAPI session authentication
+        logging.info("Using XOLite mode (XAPI session authentication)")
+
+        # Get credentials from args or environment variables
+        username = args.user if args.user else os.environ.get('HOST_DEFAULT_USER', 'root')
+        password = args.password if args.password else os.environ.get('HOST_DEFAULT_PASSWORD', '')
+
+        # Output filename from positional arg or default
+        output_file = args.url_or_filename if args.url_or_filename else "screenshot.png"
+
+        logging.info(f"Authenticating with XAPI at {args.host}...")
+        try:
+            session_id = xapi_authenticate(args.host, username, password)
+            logging.info(f"Authentication successful, session ID: {session_id[:20]}...")
+        except Exception as e:
+            print(f"Authentication failed: {e}")
+            sys.exit(1)
+
+        # Construct WebSocket URL
+        ws_url = f"wss://{args.host}/console?uuid={args.vm_uuid}&session_id={session_id}"
+        cookie = None
+
+    else:
+        # Neither mode specified - show nice usage
+        print("VNC WebSocket Console Capture")
```

Comment:
This should also probably be generated by argparse

Reply:
Done